curweather/curweather.php

```diff
@@ -12,84 +12,97 @@
 require_once 'mod/proxy.php';
 require_once 'include/text.php';
 
+use Friendica\App;
 use Friendica\Core\Addon;
 use Friendica\Core\Cache;
 use Friendica\Core\Config;
 use Friendica\Core\L10n;
 use Friendica\Core\PConfig;
 use Friendica\Util\Network;
 
+function curweather_install()
+{
+	Addon::registerHook('network_mod_init'   , 'addon/curweather/curweather.php', 'curweather_network_mod_init');
+	Addon::registerHook('addon_settings'     , 'addon/curweather/curweather.php', 'curweather_addon_settings');
+	Addon::registerHook('addon_settings_post', 'addon/curweather/curweather.php', 'curweather_addon_settings_post');
+}
+
+function curweather_uninstall()
+{
+	Addon::unregisterHook('network_mod_init'   , 'addon/curweather/curweather.php', 'curweather_network_mod_init');
+	Addon::unregisterHook('addon_settings'     , 'addon/curweather/curweather.php', 'curweather_addon_settings');
+	Addon::unregisterHook('addon_settings_post', 'addon/curweather/curweather.php', 'curweather_addon_settings_post');
+}
+
 //  get the weather data from OpenWeatherMap
-function getWeather( $loc, $units='metric', $lang='en', $appid='', $cachetime=0) {
-	$url = "http://api.openweathermap.org/data/2.5/weather?q=".$loc."&appid=".$appid."&lang=".$lang."&units=".$units."&mode=xml";
+function getWeather($loc, $units = 'metric', $lang = 'en', $appid = '', $cachetime = 0)
+{
+	$url = "http://api.openweathermap.org/data/2.5/weather?q=" . $loc . "&appid=" . $appid . "&lang=" . $lang . "&units=" . $units . "&mode=xml";
 	$cached = Cache::get('curweather'.md5($url));
 	$now = new DateTime();
+
 	if (!is_null($cached)) {
 		$cdate = PConfig::get(local_user(), 'curweather', 'last');
 		$cached = unserialize($cached);
+
 		if ($cdate + $cachetime > $now->getTimestamp()) {
 			return $cached;
 		}
 	}
 	try {
 		$res = new SimpleXMLElement(Network::fetchUrl($url));
 	} catch (Exception $e) {
-		if (!$_SESSION['curweather_notice_shown']) {
+		if (empty($_SESSION['curweather_notice_shown'])) {
 			info(L10n::t('Error fetching weather data. Error was: '.$e->getMessage()));
 			$_SESSION['curweather_notice_shown'] = true;
 		}
+
 		return false;
 	}
+
 	unset($_SESSION['curweather_notice_shown']);
-	if ((string)$res->temperature['unit']==='metric') {
+
+	if ((string) $res->temperature['unit'] === 'metric') {
 		$tunit = '°C';
 		$wunit = 'm/s';
 	} else {
 		$tunit = '°F';
 		$wunit = 'mph';
 	}
-	if ( trim((string)$res->weather['value']) == trim((string)$res->clouds['name']) ) {
-		$desc = (string)$res->clouds['name'];
+
+	if (trim((string) $res->weather['value']) == trim((string) $res->clouds['name']) ) {
```

Comment:
There still is an extra space at the end of the condition.

Reply:
This is still valid after the latest commit.

Reply:
Now finally found. :-)